apprise/plugins/one_signal.py

```diff
@@ -317,6 +364,12 @@ def send(self, body, title='', notify_type=NotifyType.INFO, **kwargs):
         if self.template_id:
             payload['template_id'] = self.template_id
 
+        if self.template_data:
+            payload["custom_data"] = self.template_data
+
+        if self.onesignal_data:
```

Comment:
I don't see where this is documented in [the API documentation](https://documentation.onesignal.com/reference/create-notification).  What does the `data` tag do?

Reply:
It's easy to confuse those: `custom_data` is used to template the notification body and title, whereas `data` is posted back to their services (and then via Onesignal webhooks back to the original sending application). The latter is used, for example, to analyze notification's performance in post based on some grouping identifier, for example it could be like `{"data": {"send_id": "123"}}`, which you then see all the individual notifications with `send_id` attached to each of them, then you can group by it for analysis purposes.
This is documented [here](https://documentation.onesignal.com/reference/push-channel-properties#attachments), as I mentioned in a comment above.